File: teste2.py
def soma(a, b):
    return a + b

def fibonacci (num):
    if num == 0:
        return 0
    elif num == 1:
        return 1
    else:
        return soma(fibonacci(num-1), fibonacci(num-2))
    
def verifica_fibonacci(num):
    if num < 0:
        return False
    else:
        i = 0
        while fibonacci(i) <= num:
            if fibonacci(i) == num:
                return True
            i += 1
        return False

def main():
    num = int(input("Digite um número para verificar se ele pertence à sequência de Fibonacci: "))

    if num < 0:
        print("Número inválido")
    else:
        if verifica_fibonacci(num):
            print("O número", num, "pertence à sequência de Fibonacci")
        else:
            print("O número", num, "não pertence à sequência de Fibonacci")

if __name__ == "__main__":
    main()

File: teste2.js
const fibonacci = (n) => {
    if (n === 0) return 0;
    if (n === 1) return 1;
    return fibonacci(n - 1) + fibonacci(n - 2);
    };

const verify_fibonacci = (n) => {
    if (n < 0) return false;
    let i = 0;
    while (fibonacci(i) <= n) {
        if (fibonacci(i) === n) return true;
        i++;
        }
    return false;
    }

const isFibonnaci = (num) => {
    if (verify_fibonacci(num)) {
        console.log(num + " é um número de Fibonacci");
    } else {
        console.log(num + " não é um número de Fibonacci");
    }
}


isFibonnaci(5);
isFibonnaci(55);
isFibonnaci(89);
isFibonnaci(200);
isFibonnaci(44);
